frontend/src/HelperFunctions/customAxios.js

```diff
@@ -0,0 +1,23 @@
+import axios from "axios";
+import history from "./customHistory";
+
+const axiosInstance = axios.create();
+
+axiosInstance.interceptors.response.use(
+  function (response) {
+    return response;
+  },
+  function (er) {
+    if (axios.isAxiosError(er)) {
+      if (er.response) {
+        if (er.response.status == 401) {
+          history.replace("/login");
+        }
+      }
+    }
```

Comment:
Saves you a bit of nesting. You can use `foo?.bar` to say `if foo is undefined, don't try to access bar`. Very handy for these kinds of logic.

```suggestion
      if (axios.isAxiosError(er) && er.response?.status == 401) {
          history.replace("/login");
      }
```

Reply:
Done.